test.js

```diff
@@ -522,3 +523,60 @@ test('throw if creating child without bindings', function (t) {
     instance.child()
   })
 })
+
+test('extreme mode', function (t) {
+  var now = Date.now
+  var hostname = os.hostname
+  var proc = process
+  global.process = {
+    __proto__: process,
+    pid: 123456
+  }
+  Date.now = function () {
+    return 1459875739796
+  }
+  os.hostname = function () {
+    return 'abcdefghijklmnopqr'
+  }
+  delete require.cache[require.resolve('./')]
+  var pino = require('./')
+  var expected = ''
+  var actual = ''
+  var normal = pino(writeStream(function (s, enc, cb) {
+    expected += s
+    cb()
+  }))
+
+  var extreme = pino({extreme: true}, writeStream(function (s, enc, cb) {
+    actual += s
+    cb()
+  }))
+
+  var i = 44
+  while (i--) {
+    normal.info('h')
+    extreme.info('h')
+  }
+
+  var expected2 = expected.split('\n')[0]
+  var actual2 = ''
+
+  var e = 'global.process = { __proto__: process,  pid: 123456};Date.now = function () { return 1459875739796;};require("os").hostname = function () { return "abcdefghijklmnopqr";};var pino = require("./");var extreme = pino({extreme: true});extreme.info("h")'
+
+  var child = spawn('node', ['-e', e])
+  child.stdout.pipe(writeStream(function (s, enc, cb) {
+    actual2 += s
+    cb()
+  }))
+
+  child.on('close', function () {
+    t.is(actual, expected)
+    t.is(actual2.trim(), expected2)
+
+    os.hostname = hostname
+    Date.now = now
+    global.process = proc
```

Comment:
we are using `tap`, so you can do `t.tearDown(func)` so we are 100% sure that these get emitted correctly.